build failure with libressl 2.5.3

When building with libressl 2.5.3 on alpine linux:

````shell
gcc -O2 -std=gnu99 -fPIC -g -Wall -Wextra  -Wno-missing-field-initializers  -Wno-override-init -Wno-unused -Os -fomit-frame-pointer -DHAVE_EVENTFD=1 -DHAVE_EPOLL_CREATE1=1 -DHAVE_PIPE2=1 -D_REENTRANT -D_THREAD_SAFE -D_GNU_SOURCE -Os -fomit-frame-pointer -DSOCKET_DEBUG -DHAVE_CONFIG_H -DDNS_RANDOM=RAND_bytes -c -o /home/ncopa/aports/main/lua-cqueues/src/cqueues-rel-20161215/src/lib/notify.o /home/ncopa/aports/main/lua-cqueues/src/cqueues-rel-20161215/src/lib/notify.c
/home/ncopa/aports/main/lua-cqueues/src/cqueues-rel-20161215/src/lib/socket.c: In function 'compat_SSL_is_server':
/home/ncopa/aports/main/lua-cqueues/src/cqueues-rel-20161215/src/lib/socket.c:193:13: error: 'SSL_METHOD {aka const struct ssl_method_st}' has no member named 'ssl_connect'
  if (!method->ssl_connect || method->ssl_connect == SSLv23_server_method()->ssl_connect)
             ^~
/home/ncopa/aports/main/lua-cqueues/src/cqueues-rel-20161215/src/lib/socket.c:193:36: error: 'SSL_METHOD {aka const struct ssl_method_st}' has no member named 'ssl_connect'
  if (!method->ssl_connect || method->ssl_connect == SSLv23_server_method()->ssl_connect)
                                    ^~
/home/ncopa/aports/main/lua-cqueues/src/cqueues-rel-20161215/src/lib/socket.c:193:75: error: 'SSL_METHOD {aka const struct ssl_method_st}' has no member named 'ssl_connect'
  if (!method->ssl_connect || method->ssl_connect == SSLv23_server_method()->ssl_connect)
````

--- a/src/lib/socket.c
+++ b/src/lib/socket.c
@@ -172,6 +172,15 @@ int socket_v_api(void) {
 #define SSL_is_server(ssl) compat_SSL_is_server(ssl)
 
 static _Bool compat_SSL_is_server(SSL *ssl) {
+#if defined LIBRESSL_VERSION_NUMBER && LIBRESSL_VERSION_NUMBER >= 0x2050100fL
+	/*
+	 * NOTE: LibreSSL 2.5.1 made the SSL structure partially opaque
+	 * without providing the SSL_is_server accessor.
+	 * See libressl mailing list thread '[PATCH] add SSL_is_server'
+	 * https://marc.info/?t=149253819800003
+	 */
+	return ssl->server;
+#else
 	const SSL_METHOD *method = SSL_get_ssl_method(ssl);
 
 	/*
@@ -194,6 +203,7 @@ static _Bool compat_SSL_is_server(SSL *ssl) {
 		return 1;
 
 	return 0;
+#endif
 } /* compat_SSL_is_server() */
 #endif
 

